source/enc_ctx.c

```diff
@@ -64,29 +65,25 @@ int aws_cryptosdk_enc_ctx_size(size_t *size, const struct aws_hash_table *enc_ct
 
 int aws_cryptosdk_enc_ctx_serialize(
     struct aws_allocator *alloc, struct aws_byte_buf *output, const struct aws_hash_table *enc_ctx) {
-    size_t length;
-    if (aws_cryptosdk_enc_ctx_size(&length, enc_ctx)) {
-        return AWS_OP_ERR;
-    }
+    size_t num_elems = aws_hash_table_get_entry_count(enc_ctx);
+    if (num_elems > UINT16_MAX) return aws_raise_error(AWS_CRYPTOSDK_ERR_LIMIT_EXCEEDED);
 
-    if (output->capacity < length) {
+    if (!aws_byte_buf_write_be16(output, (uint16_t)num_elems)) {
+        aws_byte_buf_clean_up(output);
         return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
     }
 
-    if (length == 0) {
-        // Empty encryption context
-        return AWS_OP_SUCCESS;
-    }
+    if (num_elems == 0) return AWS_OP_SUCCESS;
```

Comment:
If the num_elems is zero we want to add a value 0 to the header indicating that there are no elements. Line 71-72 is initializing  `output` to 0. This needs to be added only if the num_elems is 0 though.

Reply:
Line 71-72 is writing whatever the ```num_elems``` happens to be, whether that is ```0``` or non-zero.  If its zero, we've written our 0 and are done; if it is non-zero, we also, need to output the pairs themselves, which is what the loop below does.

Reply:
sgtm. 
Could you run the format script on this? 
Our styling for the if statement is different from this one.